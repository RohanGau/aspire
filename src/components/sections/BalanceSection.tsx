import styles from './BalanceSection.module.scss';
import { useState } from 'react';

interface BalanceSectionProps {
  balance: number;
  currency: string;
}

const BalanceSection = ({ balance, currency }: BalanceSectionProps) => {
  const [visible, setVisible] = useState(true);

  return (
    <div className={styles.balanceCard}>
      <div className={styles.balanceCard__header}>
        <p className={styles.balanceCard__label}>Available Balance</p>
        <button
          className={styles.balanceCard__toggle}
          onClick={() => setVisible((prev) => !prev)}
          aria-label="Toggle Balance Visibility"
        ></button>
      </div>

      <div className={styles.balanceCard__amount}>
        {visible ? (
          <h2>
            {currency} {balance.toLocaleString()}
          </h2>
        ) : (
          <h2>****</h2>
        )}
      </div>

      <button className={styles.balanceCard__cta}>{/* <FaPlus /> New Card */}</button>
    </div>
  );
};

export default BalanceSection;
