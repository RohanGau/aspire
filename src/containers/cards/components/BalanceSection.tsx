import styles from './BalanceSection.module.scss';
import { Skeleton } from 'antd';

interface BalanceSectionProps {
  isLoading: boolean;
  balance: number;
  currency: string;
  onhandleAddCard: () => void;
}

const BalanceSection = ({ isLoading, balance, currency, onhandleAddCard }: BalanceSectionProps) => {
  return (
    <div className={styles.balanceCard}>
      <div className={styles.balanceCard__header}>
        <p className={styles.balanceCard__label}>Available Balance</p>
        <div className={styles.balanceCard__priceContainer}>
          <span className={styles.balanceCard__currency}>{currency}</span>
          {isLoading ? (
            <Skeleton.Button active size={'default'} shape={'square'} />
          ) : (
            <h2 className={styles.balanceCard__currencyText}>{balance.toLocaleString()}</h2>
          )}
        </div>
      </div>

      <div className={styles.balanceCard__buttonContainer}>
        <button className={styles.balanceCard__cta} onClick={onhandleAddCard}>
          New Card
        </button>
      </div>
    </div>
  );
};

export default BalanceSection;
