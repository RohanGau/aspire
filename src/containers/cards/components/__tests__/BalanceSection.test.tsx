import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
// import BalanceSection from './BalanceSection';
import BalanceSection from '../BalanceSection';
import { Skeleton } from 'antd';

describe('BalanceSection', () => {
  const mockHandleAddCard = jest.fn();
  const defaultProps = {
    isLoading: false,
    balance: 12345.67,
    currency: 'S$',
    onhandleAddCard: mockHandleAddCard,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders loading skeleton when isLoading is true', () => {
    render(<BalanceSection {...defaultProps} isLoading={true} />);

    // Check for Ant Design Skeleton component
    const skeleton = document.querySelector('.ant-skeleton-button');
    expect(skeleton).toBeInTheDocument();
    expect(skeleton).toHaveClass('ant-skeleton-button');
  });

  it('displays balance and currency correctly when not loading', () => {
    render(<BalanceSection {...defaultProps} />);

    expect(screen.getByText('Available Balance')).toBeInTheDocument();
    expect(screen.getByText('S$')).toBeInTheDocument();
    expect(screen.getByText('12,345.67')).toBeInTheDocument();
  });

  it('calls onhandleAddCard when "New Card" button is clicked', () => {
    render(<BalanceSection {...defaultProps} />);

    const button = screen.getByRole('button', { name: /new card/i });
    fireEvent.click(button);

    expect(mockHandleAddCard).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot when loading', () => {
    const { asFragment } = render(<BalanceSection {...defaultProps} isLoading={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when not loading', () => {
    const { asFragment } = render(<BalanceSection {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
