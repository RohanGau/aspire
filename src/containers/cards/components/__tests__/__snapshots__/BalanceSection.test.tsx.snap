// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceSection matches snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="balanceCard"
  >
    <div
      class="balanceCard__header"
    >
      <p
        class="balanceCard__label"
      >
        Available Balance
      </p>
      <div
        class="balanceCard__priceContainer"
      >
        <span
          class="balanceCard__currency"
        >
          S$
        </span>
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-1v5z42l"
        >
          <span
            class="ant-skeleton-button ant-skeleton-button-square"
          />
        </div>
      </div>
    </div>
    <div
      class="balanceCard__buttonContainer"
    >
      <button
        class="balanceCard__cta"
      >
        New Card
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BalanceSection matches snapshot when not loading 1`] = `
<DocumentFragment>
  <div
    class="balanceCard"
  >
    <div
      class="balanceCard__header"
    >
      <p
        class="balanceCard__label"
      >
        Available Balance
      </p>
      <div
        class="balanceCard__priceContainer"
      >
        <span
          class="balanceCard__currency"
        >
          S$
        </span>
        <h2
          class="balanceCard__currencyText"
        >
          12,345.67
        </h2>
      </div>
    </div>
    <div
      class="balanceCard__buttonContainer"
    >
      <button
        class="balanceCard__cta"
      >
        New Card
      </button>
    </div>
  </div>
</DocumentFragment>
`;
